.github/workflows/dvc-diff.yml

```diff
@@ -0,0 +1,91 @@
+# Checks for image diffs in a Pull Request and adds a GitHub comment showing the diff
+name: DVC image diff
+
+on:
+  pull_request:
+    paths:
+      - 'pygmt/tests/baseline/*.png.dvc'
+
+jobs:
+  dvc-diff:
+    name: DVC image diff
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        # fetch all history so that git diff works
+        fetch-depth: 0
+
+    - name: Setup data version control (DVC)
+      uses: iterative/setup-dvc@v1.0.2
+
+    - name: Setup continuous machine learning (CML)
+      uses: iterative/setup-cml@v1.0.0
+
+    # Produce the markdown diff report, which should look like:
+    # ## Summary of changed images
+    #
+    # This is an auto-generated report of images that have changed on the DVC remote
+    #
+    # | Status   | Path                                |
+    # |----------|-------------------------------------|
+    # | added    | pygmt/tests/baseline/test_image.png |
+    - name: Put list of images that were added or changed into report
+      run: |
+        echo -e "## Summary of changed images\n" > report.md
+        echo -e "This is an auto-generated report of images that have changed on the DVC remote\n" >> report.md
+        dvc diff --show-md master HEAD >> report.md
```

Comment:
Could you also add the commit hash to the report? It would be easier to know which commit is the last one that updates the comment.

Reply:
> Could you also add the commit hash to the report? It would be easier to know which commit is the last one that updates the comment.

Hmm, I tried using `$GITHUB_SHA` at 6fa103f19be1f9a53a0f1705ac76da92ce7a35a7 (following https://github.com/GenericMappingTools/pygmt/commit/9b2d3efda63e86ca20efec0c9644db10ccfa5054), but the report uses https://github.com/GenericMappingTools/pygmt/commit/9b2d3efda63e86ca20efec0c9644db10ccfa5054 instead which looks like a hidden merge of 6fa103f19be1f9a53a0f1705ac76da92ce7a35a7 into the target master branch. Do you know of a way of getting the 6fa103f19be1f9a53a0f1705ac76da92ce7a35a7 hash automatically?

Reply:
```
${{ github.event.pull_request.head.sha }}
```
This one should work.